File: typescript-client-sdk/src/sdk/models/operations/getbatchcrmtask.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetBatchCrmTaskQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ids" })
  ids?: string;
}


export class GetBatchCrmTaskResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=invalidIds" })
  invalidIds?: string[];

  @SpeakeasyMetadata({ data: "json, name=tasks", elemType: shared.Task })
  tasks?: shared.Task[];
}


export class GetBatchCrmTaskRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetBatchCrmTaskQueryParams;
}


export class GetBatchCrmTaskResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetBatchCrmTaskResponseBody;
}

File: go-client-sdk/pkg/models/operations/getbatchcrmtask.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetBatchCrmTaskQueryParams struct {
	AccessToken string  `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool   `queryParam:"style=form,explode=true,name=allFields"`
	Ids         *string `queryParam:"style=form,explode=true,name=ids"`
}

type GetBatchCrmTaskResponseBody struct {
	InvalidIds []string      `json:"invalidIds,omitempty"`
	Tasks      []shared.Task `json:"tasks,omitempty"`
}

type GetBatchCrmTaskRequest struct {
	QueryParams GetBatchCrmTaskQueryParams
}

type GetBatchCrmTaskResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetBatchCrmTaskResponseBody
}
